tests/e2e/subscription_suite_test.go

```diff
@@ -39,16 +41,57 @@ var _ = ginkgov2.Describe("Subscription Config Suite Test", func() {
 			gomega.Expect(err).To(gomega.BeNil())
 			gomega.Eventually(s.CsvIsInstalling, time.Minute*1, time.Second*5).WithArguments(runTimeClientForSuiteRun).Should(gomega.BeTrue())
 
-			// get csv from installplan from subscription
-			log.Printf("Wait for CSV to be succeeded")
 			if testCase.failureExpected != nil && *testCase.failureExpected {
-				gomega.Consistently(s.CsvIsReady, time.Minute*2, time.Second*5).WithArguments(runTimeClientForSuiteRun).Should(gomega.BeFalse())
-				// TODO read error message instead?
+				haveErrorMessage := func() bool {
+					controllerManagerPods, err := kubernetesClientForSuiteRun.CoreV1().Pods(namespace).List(context.Background(), v1.ListOptions{LabelSelector: "control-plane=controller-manager"})
+					if err != nil {
+						return false
+					}
+					for _, pod := range controllerManagerPods.Items {
+						podLogs, err := lib.GetPodContainerLogs(kubernetesClientForSuiteRun, namespace, pod.Name, "manager")
+						if err != nil {
+							return false
+						}
+						if strings.Contains(podLogs, "error setting privileged pod security labels to operator namespace") && strings.Contains(podLogs, "connect: connection refused") {
+							log.Printf("found error message in controller manager Pod")
+							return true
+						}
+					}
+					return false
+				}
+				gomega.Eventually(haveErrorMessage, time.Minute*1, time.Second*5).Should(gomega.BeTrue())
 			} else {
-				gomega.Eventually(s.CsvIsReady, time.Minute*15, time.Second*5).WithArguments(runTimeClientForSuiteRun).Should(gomega.BeTrue())
+				log.Printf("Wait for CSV to be succeeded")
+				gomega.Eventually(s.CsvIsReady, time.Minute*7, time.Second*5).WithArguments(runTimeClientForSuiteRun).Should(gomega.BeTrue())
+
+				var controllerManagerPodName string
+				onlyOneManagerPod := func() bool {
+					controllerManagerPod, err := kubernetesClientForSuiteRun.CoreV1().Pods(namespace).List(context.Background(), v1.ListOptions{LabelSelector: "control-plane=controller-manager"})
+					if err != nil {
+						return false
+					}
+					log.Printf("waiting for having only one controller manager Pod")
+					if len(controllerManagerPod.Items) == 1 {
+						controllerManagerPodName = controllerManagerPod.Items[0].Name
+						return true
+					}
+					return false
+				}
+
+				gomega.Eventually(onlyOneManagerPod, time.Minute*4, time.Second*5).Should(gomega.BeTrue())
+
+				isLeaseReady := func() bool {
+					podLogs, err := lib.GetPodContainerLogs(kubernetesClientForSuiteRun, namespace, controllerManagerPodName, "manager")
+					if err != nil {
+						return false
+					}
+					log.Printf("waiting leaderelection")
+					return strings.Contains(podLogs, "leaderelection.go:258] successfully acquired lease")
+				}
+				gomega.Eventually(isLeaseReady, time.Minute*4, time.Second*5).Should(gomega.BeTrue())
 
 				log.Printf("Creating test Velero")
-				err := dpaCR.Build(lib.CSI)
+				err = dpaCR.Build(lib.CSI)
```

Comment:
Why are we writing to err outside the else block?
What does this change do?

Reply:
I think in my early implementations I was creating error before that line, then removed here

it could be reverted, but if harmless, would let it be to not push again

Reply:
harmless enough, nothing else reading it after the block atm.